pkg/server/sessions/internal/transactions/transactions.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 CodeNotary, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transactions
+
+import (
+	"github.com/codenotary/immudb/embedded/sql"
+	"github.com/codenotary/immudb/pkg/api/schema"
+	"github.com/codenotary/immudb/pkg/database"
+	"sync"
+)
+
+type transaction struct {
+	sync.Mutex
+	transactionID string
+	sqlTx         *sql.SQLTx
+	txMode        schema.TxMode
+	db            database.DB
+	sessionID     string
+}
+
+type Transaction interface {
+	GetID() string
```

Comment:
Database getter may be useful for the future as well

Reply:
yes, at the moment is not used. When it will be needed I will add